contrib/orafce/charpad.c

```diff
@@ -287,17 +320,47 @@ orafce_rpad(PG_FUNCTION_ARGS)
 	bool	s2_operate = ON,
 			half_space = OFF,
 			init_ptr = ON;
+	int		nargs;
+	Oid		argoid1,argoid2,argoid3;
+
+	nargs = PG_NARGS();
+	if (nargs < 2)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				errmsg("too few args.")));
+	if (nargs > 3)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				errmsg("more args than function needed.")));
```

Comment:
ERRCODE_TOO_MANY_ARGUMENTS is more appropriate here. and the error message can also like this.
```suggestion
	if (nargs > 3)
 		ereport(ERROR,
				(errcode(ERRCODE_TOO_MANY_ARGUMENTS),
				 errmsg("too many arguments")));
```